tools/serverpod_cli/lib/src/generator/dart/library_generators/server_test_tools_generator.dart

```diff
@@ -173,7 +173,9 @@ class ServerTestToolsGenerator {
                 'methodName': literalString(method.name),
                 'parameters': literalMap({
                   for (var parameter in method.allParameters)
-                    literalString(parameter.name): refer(parameter.name).code,
+                    literalString(parameter.name):
+                        refer('testObjectToJson', serverpodTestUrl)
+                            .call([refer(parameter.name)]).code,
```

Comment:
Question: Why can't we just encode the parameters the same way as we do in the client?

https://github.com/serverpod/serverpod/blob/3b1594b732fd8c49a0fe3d6ccfaa6db1271ae6bf/packages/serverpod_client/lib/src/serverpod_client_shared_private.dart#L8-L15

Reply:
This might not be a problem but I suspect this should actually be `.encodeForProtocol` right?

Reply:
> This might not be a problem but I suspect this should actually be `.encodeForProtocol` right?

I guess so? 🧐

Reply:
OK! Makes sense, can we create a general method that handles all parameters instead of handling them specifically?

Reply:
@SandPod Fixed: [3433cc8](https://github.com/serverpod/serverpod/pull/2846/commits/3433cc81c5a5fd11d1c9c0451c6a0c915ba6a9c9) 👍